packages/vscode-extension/src/network/App.tsx

```diff
@@ -27,58 +43,65 @@ function App() {
     handleResize.flush();
 
     window.addEventListener("resize", handleResize);
+
     return () => {
       window.removeEventListener("resize", handleResize);
       handleResize.cancel();
     };
   }, []);
 
-  const { networkLogs } = useNetwork();
+  useEffect(() => {
+    // Set the size of the network log details container, after users decides to resize it
+    // https://vscode-elements.github.io/components/split-layout/api/
+    const handleResize = debounce(() => {
+      if (!networkLogContainerRef.current || !networkDetailsContainerRef.current) {
+        return;
+      }
+      const containerWidth = networkLogContainerRef.current.clientWidth;
+      const detailsWidth = networkDetailsContainerRef.current?.clientWidth;
+      networkLogDetailsSize.current = `${((containerWidth - detailsWidth) / containerWidth) * 100}%`;
+    });
 
-  const [selectedNetworkLogId, setSelectedNetworkLogId] = useState<string | null>(null);
+    const detailsResizeObserver = new ResizeObserver(handleResize);
 
-  const selectedNetworkLog = useMemo(() => {
-    const fullLog = networkLogs.find((log) => log.requestId === selectedNetworkLogId);
-    if (!fullLog) {
-      setSelectedNetworkLogId(null);
+    if (networkDetailsContainerRef.current) {
+      detailsResizeObserver.observe(networkDetailsContainerRef.current);
     }
-    return fullLog || null;
-  }, [selectedNetworkLogId, networkLogs]);
 
-  const isNetworkLogDetailsVisible = !!selectedNetworkLog;
+    return () => {
+      detailsResizeObserver.disconnect();
+      handleResize.cancel();
+    };
```

Comment:
```suggestion
    if (!networkDetailsContainerRef.current) {
      return;
    }

    const detailsResizeObserver = new ResizeObserver(handleResize);
    detailsResizeObserver.observe(networkDetailsContainerRef.current);
    return () => {
      detailsResizeObserver.disconnect();
      handleResize.cancel();
    };
```

You can skip creating the resize observer if the ref isn't available.

Reply:
Changed in cf3081e0 

```ts
useEffect(() => {
    // Set the size of the network log details container, after users decides to resize it
    // https://vscode-elements.github.io/components/split-layout/api/

    const networkDetailsContainer = networkDetailsContainerRef.current;
    if (!networkDetailsContainer) {
      return;
    }

    const handleResize = debounce(() => {
      const networkLogContainer = networkLogContainerRef.current;
      if (!networkLogContainer || !networkDetailsContainer) {
        return;
      }
      const containerWidth = networkLogContainer.clientWidth;
      const detailsWidth = networkDetailsContainer?.clientWidth;
      networkLogDetailsSize.current = `${((containerWidth - detailsWidth) / containerWidth) * 100}%`;
    }, DEBOUNCE_TIME);

    const detailsResizeObserver = new ResizeObserver(handleResize);
    detailsResizeObserver.observe(networkDetailsContainer);

    return () => {
      detailsResizeObserver.disconnect();
      handleResize.cancel();
    };
  }, [areNetworkLogDetailsVisible]);
```